ddtrace/tracer/tracer.go

```diff
@@ -66,6 +69,8 @@ type tracer struct {
 	// These integers track metrics about spans and traces as they are started,
 	// finished, and dropped
 	spansStarted, spansFinished, tracesDropped int64
+
+	payloads sync.Pool
```

Comment:
The payloads pool was just an idea I had.
It may be premature optimization, but it should help prevent repeated allocs of the payload buffer and avoid GC work.
We may want to do some benchmarking.